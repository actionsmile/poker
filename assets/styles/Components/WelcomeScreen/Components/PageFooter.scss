.mainPage {
    &--footer {
        padding: 1em;
        margin: 0;

        a.level-item {
            border: none;
        }
    }
}
