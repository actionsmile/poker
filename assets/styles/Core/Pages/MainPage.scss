.mainPage {
    &--container {
        padding: 1em;
    }
}