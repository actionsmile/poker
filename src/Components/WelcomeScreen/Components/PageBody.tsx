import * as React from 'react';

export class PageBody extends React.Component {
    public render(): JSX.Element {
        return (
            <div className="hero-body"></div>
        );
    }
}
