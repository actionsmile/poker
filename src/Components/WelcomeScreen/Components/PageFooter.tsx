import * as React from 'react';

export class PageFooter extends React.Component {

    public render(): JSX.Element {
        return (
            <div className="hero-foot mainPage--footer">
                <div className="level">
                    <div className="level-left"></div>
                    <div className="level-right">
                        <a href="https://bulma.io" className="level-item">
                            <img src="https://bulma.io/images/made-with-bulma--white.png" alt="Made with Bulma" width="128" height="24" />
                        </a>
                    </div>
                </div>
            </div>
        );
    }
}
