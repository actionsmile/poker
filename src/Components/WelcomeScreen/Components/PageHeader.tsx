import * as React from 'react';

export class PageHeader extends React.Component {

    public render(): JSX.Element {
        return (
            <div className="hero-head"></div>
        );
    }
}
