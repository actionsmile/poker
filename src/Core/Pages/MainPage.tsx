import * as React from 'react';
import { connect } from 'react-redux';
import { IAppState, IGreetingsState } from 'Core/Models';

interface IMainPageProps {
    greetings: IGreetingsState;
}

class MainPageView extends React.Component<IMainPageProps> {
    private handleClick = (): void => {
        console.log('Click!');
    };

    public render(): JSX.Element {
        const { greetings } = this.props;
        return (
            <div className="hero is-dark is-fullheight">
                <div className="hero-body">
                    <div className="container">
                        <h1 className="title">{greetings.title}</h1>
                        <button
                            className="button is-outlined is-info is-rounded subtitle"
                            onClick={this.handleClick}
                        >
                            {greetings.home ? greetings.home.label : 'Hmâ€¦'}
                        </button>
                    </div>
                </div>
                <div className="hero-foot mainPage--footer">
                    <div className="level">
                        <div className="level-left"></div>
                        <div className="level-right">
                            <a href="https://bulma.io" className="level-item">
                                <img src="https://bulma.io/images/made-with-bulma--white.png" alt="Made with Bulma" width="128" height="24" />
                            </a>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state: IAppState): IMainPageProps => ({
    greetings: state.greetings
});

export const MainPage = connect<{}, IMainPageProps, {}>(mapStateToProps, null)(
    MainPageView
);
