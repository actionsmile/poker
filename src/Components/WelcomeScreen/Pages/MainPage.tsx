import * as React from 'react';
import { PageFooter } from './../Components/PageFooter';

export class MainPage extends React.Component {

    public render(): JSX.Element {
        return (
            <div className="hero is-dark is-fullheight">
                <div className="hero-body"></div>
                <PageFooter />
            </div>
        );
    }
}
