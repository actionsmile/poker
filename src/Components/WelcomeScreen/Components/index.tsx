import { PageBody } from './PageBody';
import { PageFooter } from './PageFooter';
import { PageHeader } from './PageHeader';

export const Body = PageBody;
export const Footer = PageFooter;
export const Header = PageHeader;
