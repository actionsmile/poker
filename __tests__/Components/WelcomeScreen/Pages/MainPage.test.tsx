import * as React from 'react';
import { create } from 'react-test-renderer';
import { PageFooter } from 'Components/WelcomeScreen/Components/PageFooter';
import { MainPage } from 'Components/WelcomeScreen/Pages/MainPage'

describe('MainPage component', () => {
    const wrapper = create(<MainPage />);
    const mainPage = wrapper.root;
    it('MainPage renders without crashes', () => {
        expect(mainPage.type).toBe(MainPage);
    });

    it('Contains just one <PageFooter /> element', () => {
        expect(mainPage.findAllByType(PageFooter).length).toEqual(1);
    });
});
