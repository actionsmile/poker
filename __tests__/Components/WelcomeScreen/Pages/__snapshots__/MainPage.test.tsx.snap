// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component Matches Snapshot 1`] = `
<div
  className="hero is-dark is-fullheight"
>
  <div
    className="hero-head"
  />
  <div
    className="hero-body"
  />
  <div
    className="hero-foot mainPage--footer"
  >
    <div
      className="level"
    >
      <div
        className="level-left"
      />
      <div
        className="level-right"
      >
        <a
          className="level-item"
          href="https://bulma.io"
        >
          <img
            alt="Made with Bulma"
            height="24"
            src="https://bulma.io/images/made-with-bulma--white.png"
            width="128"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
